sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs

```diff
@@ -467,7 +472,8 @@ private static void CloseLink(RequestResponseAmqpLink link)
                 var i = 0;
                 foreach (ArraySegment<byte> deliveryTag in deliveryTags)
```

Comment:
Do we ever actually send multiple lock tokens?   Unless I'm missing something, there seems to be only a single one wrapped in an array at each call site.  This seems to be something that we call frequently as part of the core flow.  Should we consider an overload optimized for a single token?

Reply:
We never actually call this with multiple tokens. Considering this functionality would no longer be needed the best way would be to remove the multiple token approach. This would also get rid of a lot of List and Array allocs entirely. I've pointed this out in the [issue ](https://github.com/Azure/azure-sdk-for-net/issues/20166)linked in the description.

> The code only ever passes one lock token in so by getting rid of all the enumeration we could stackalloc the required memory for the guid and then marshal it into. This would also get rid of all the task lists etc and make the code in general more efficient or if the multiple lock token path needs to be preserved

But since nobody answered there, and I didn't know if the code is still needed or not I figured it is best to assume it is and went ahead with this approach. I've also considered having overloads but then I thought the redundancy involved might also be questioned during the reviews so I stuck to this one for now. Happy to change

Reply:
![image](https://user-images.githubusercontent.com/174258/114895222-9bc4dc00-9e0f-11eb-89f5-865307a41160.png)

Reply:
Unfortunately, I don't have good insight into some of the nuances around the lock management nor potential future scenarios.   @JoshLove-msft would definitely be the better option to make the call.

Reply:
Writing the alternative took me less than 20 min so I just went ahead and sent in a second proposal

https://github.com/Azure/azure-sdk-for-net/pull/20427

Reply:
Alternative approach looks good.

Reply:
So close this one?

Reply:
Yep, I think so.